onnx/shape_inference/implementation.cc

```diff
@@ -371,6 +372,38 @@ void InferShapes(
       m.ir_version());
 }
 
+void InferShapes(
+  const std::string& model_path,
+  const bool check_type,
+  const ISchemaRegistry* schema_registry
+  ) {
+  ModelProto model;
+  std::fstream model_stream(model_path, std::ios::in | std::ios::binary);
+  if (!model_stream.good()) {
+    fail_check(
+        "Unable to open model file:",
+        model_path,
+        ". Please check if it is a valid file.");
+  }
+  std::string data{std::istreambuf_iterator<char>{model_stream},
+                   std::istreambuf_iterator<char>{}};
+  if (!ParseProtoFromBytes(&model, data.c_str(), data.size())) {
+    fail_check(
+        "Unable to parse model from file:",
+        model_path,
+        ". Please check if it is a valid protobuf file of model.");
+  }
+  InferShapes(model, check_type, schema_registry);
+  // Save the inferred model to the original model path
+  // TODO catch fail status
+  std::fstream output(model_path, std::ios::out | std::ios::trunc | std::ios::binary);
+  if (!model.SerializeToOstream(&output)) {
```

Comment:
Would it be better to support a separate input_model_path and a separate output_model_path?

Reply:
Sounds good. We can add an optional arguments, `output_model_path`, for Python API. And I think the `output_model_path` needs to be in the same directory as the original model_path because this output only works with the main onnx model file... We should state this limitation for the `output_model_path`.